apps/emqx/test/emqx_common_test_helpers.erl

```diff
@@ -1007,7 +1007,11 @@ clear_screen() ->
     end.
 
 with_mock(Mod, FnName, MockedFn, Fun) ->
-    ok = meck:new(Mod, [no_link, no_history, passthrough]),
+    with_mock(Mod, FnName, MockedFn, _Opts = #{}, Fun).
+
+with_mock(Mod, FnName, MockedFn, Opts, Fun) ->
+    MeckOpts = maps:get(meck_opts, Opts, [no_link, no_history, passthrough]),
```

Comment:
Curious, why do we use `no_link` by default? Mocking from outside a test case process seems a rare case. Also this imposes an obligation to make explicit meck cleanups.

Reply:
Good question, it was there for a long time. IIRC, when using meck in some clustered tests, this option is needed.  :thinking: 

We could change this default in another PR and then fix the cases that break.